plugins/colorschemes/base16/default.nix

```
@@ -37,94 +39,131 @@ helpers.vim-plugin.mkVimPlugin config {
       ])
     ];
 
+  settingsExample = {
+    telescope_borders = true;
+    indentblankline = false;
+    dapui = false;
+  };
+
+  settingsOptions = with helpers.defaultNullOpts; {
+    telescope = mkBool true ''
+      Whether to enable telescope integration.
+    '';
+    telescope_borders = mkBool false ''
```

```suggestion
  settingsOptions = {
    telescope = helpers.defaultNullOpts.mkBool true ''
      Whether to enable telescope integration.
    '';

    telescope_borders = helpers.defaultNullOpts.mkBool false ''
```

Two reasons:
- Consistence with the rest of the code base (very valuable IMO for new contributors)
- Following nixpkgs' recommendation of discouraging the use of `with`.

---

Agreed. Not sure why I was using `with` there in the first place!

TBH, even the way we usually put `with lib;` at the start of a file goes against best practices and nixd will be shouting `sema-escaping-with` all over the place unless you suppress that warning. But it does make the code less verbose and is consistent with the rest of the codebase...

---

> TBH, even the way we usually put `with lib;` at the start of a file goes against best practices and nixd will be shouting `sema-escaping-with` all over the place unless you suppress that warning. But it does make the code less verbose and is consistent with the rest of the codebase...

Sure, We'd have to assess the relevance of an alternative approach. But this is out of scope for this PR.